open_seq2seq/data/text2speech/text2speech.py

```diff
@@ -0,0 +1,290 @@
+    pad_to = self.params.get('pad_to', 8)
+    if self.load_from_disk:
+      file_path = os.path.join(self.params['dataset_location'],audio_filename+".npy")
+      spectrogram = np.load(file_path)
+      mag_power = self.params.get('mag_power', 2)
+      if mag_power != 1:
+        spectrogram = spectrogram * mag_power
+        spectrogram = np.clip(spectrogram, a_min=np.log(1e-5), a_max=None)
+      if self.mel:
+        spectrogram = get_mel(spectrogram, power=mag_power,
+                              feature_normalize=self.params["feature_normalize"],
+                              mean=self.params.get("feature_normalize_mean", 0.),
+                              std=self.params.get("feature_normalize_std", 1.)
+                             )
+      # Else it is a magnitude spec, and we need to normalize
+      elif self.params["feature_normalize"]:
+        spectrogram = normalize(spectrogram,
+                                mean=self.params.get("feature_normalize_mean", 0.),
+                                std=self.params.get("feature_normalize_std", 0.))
+    else:
+      file_path = os.path.join(self.params['dataset_location'],audio_filename+".wav")
+      spectrogram = get_speech_features_from_file(
+        file_path, self.params['num_audio_features'], pad_to,
+        features_type=self.params['output_type'],
+        mag_power=self.params.get('mag_power', 2),
+        feature_normalize=self.params["feature_normalize"],
+        mean=self.params.get("feature_normalize_mean", 0.),
+        std=self.params.get("feature_normalize_std", 1.)
+      )
+    if self.params.get("pad_EOS", True):
+      spectrogram = np.pad(spectrogram, ((0,1),(0,0)), "constant", constant_values=0)
+    stop_token_target = np.zeros([len(spectrogram)], dtype=self.params['dtype'].as_numpy_dtype())
+    stop_token_target[-1] = 1.
+    return np.int32(text_input), \
+           np.int32([len(text_input)]), \
+           spectrogram.astype(self.params['dtype'].as_numpy_dtype()), \
+           stop_token_target.astype(self.params['dtype'].as_numpy_dtype()), \
+           np.int32([len(spectrogram)])
+
+  def _parse_transcript_element(self, transcript):
+    """Parses text from file and returns array of text features.
+
+    Args:
+      text: the string to parse.
+
+    Returns:
+      tuple: target text as `np.array` of ids, target text length.
+    """
+    if not six.PY2:
+      transcript = str(transcript, 'utf-8')
+    transcript = transcript.lower()
+    text_input = np.array([self.params['char2idx'][c] for c in unicode(transcript,"utf-8")])
+    if self.params.get("pad_EOS", True):
+      text_input = np.append(text_input, self.params['char2idx']["~"])
+
+    return np.int32(text_input), \
+           np.int32([len(text_input)])
+
+  @property
+  def input_tensors(self):
+    return self._input_tensors
+
+  def get_size_in_samples(self):
+    """Returns the number of audio files."""
+    return len(self._files)
+
+  def inverse_mel(self, spectrogram):
```

Comment:
It's a good style to make such function private (by starting with "_") to indicate that they are only used inside some other methods of this class and should not be used by users in other places

Reply:
Oh, I want to be able to call this function from other classes. Should I modify the code style in some way to reflect this?

Reply:
Can you point me out to where you need to call it?

Reply:
https://github.com/blisc/OpenSeq2Seq/blob/73725887d711c11521b830abc737f219f9e428fc/open_seq2seq/models/text2speech.py#L298

Reply:
Alright, then let's keep it as it is.